### PR TITLE
Improve the API sections, add more classes

### PR DESCRIPTION
**Pull request by danieldk**
## Description

Improve the API sections, add more classes.  Also infer parameter type documentation from type annotations.

### Types of change

Documentation improvements.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: curated_transformers/models/embeddings.py [hunk 1]
```diff
@@ -94,10 +94,10 @@ def forward(self, input: torch.Tensor, *, positions: Optional[Tensor] = None):
         """
         Apply rotary embeddings to an array.
 
-        input (Tensor): Array to apply the rotary embeddings to.
-        positions (Optional[Tensor]): positions of the inputs. If no
-            positions are provided, they are assumed to be [0, seq_len).
-        RETURNS (Tensor): Array with the rotary embeddings applied.
+        :param input: Array to apply the rotary embeddings to.
```
Comment (danieldk) [inline curated_transformers/models/embeddings.py:97]: Had to fix this up already, because I was getting errors when generating the documentation.

Comment (shadeMe) [reply curated_transformers/models/embeddings.py:97]: This means that the docs generator will automatically pull the type and default values from the function signature?

Comment (danieldk) [reply curated_transformers/models/embeddings.py:97]: Not sure what is happening with the constructor documentation, let me check...

Comment (danieldk) [reply curated_transformers/models/embeddings.py:97]: Better:

Screenshot 2023-05-16 at 11 13 16

**Review by shadeMe — APPROVED** [review]

**Pull request merged**